src/GovUk.Education.ExploreEducationStatistics.Admin/Services/ManageContent/KeyStatisticService.cs

```diff
@@ -0,0 +1,188 @@
+    {
+        return await _persistenceHelper.CheckEntityExists<Release>(releaseId)
+            .OnSuccess(_userService.CheckCanUpdateRelease)
+            .OnSuccess(async release =>
+                await _persistenceHelper.CheckEntityExists<KeyStatistic>(keyStatisticId, query =>
+                    query
+                        .Include(keyStat => (keyStat as KeyStatisticDataBlock)!.DataBlock) // TODO Include can be removed in EES-3988
+                        .Where(keyStat => keyStat.ReleaseId == release.Id)))
+            // TODO: Can remove the below OnSuccess in EES-3988
+            // Key stats were previously associated with a KeyStatistic content section,
+            // which meant their data blocks had a ContentSectionId set. The below code
+            // ensures ContentSectionId is unset when a key stat is removed. If it isn't
+            // unset, that data block won't become available for selection again.
+            .OnSuccess(async keyStat =>
+            {
+                if (keyStat is KeyStatisticDataBlock keyStatDataBlock)
+                {
+                    var dataBlock = keyStatDataBlock.DataBlock;
+                    dataBlock.ContentSection = null;
+                    dataBlock.ContentSectionId = null;
+                    _context.ContentBlocks.Update(dataBlock);
+                    await _context.SaveChangesAsync();
+                }
+                return keyStat;
+            })
+            .OnSuccessVoid(async keyStat =>
+            {
+                _context.KeyStatistics.Remove(keyStat);
+                await _context.SaveChangesAsync();
+            });
+    }
+
+    public async Task DeleteAssociatedKeyStatisticDataBlock(Guid dataBlockId)
+    {
+        var keyStat = await _context
+            .KeyStatisticsDataBlock
+            .SingleOrDefaultAsync(ks => ks.DataBlockId == dataBlockId);
+
+        if (keyStat != null)
+        {
+            _context.KeyStatisticsDataBlock.Remove(keyStat);
+            await _context.SaveChangesAsync();
+        }
+    }
+
+    public async Task<Either<ActionResult, List<KeyStatisticViewModel>>> Reorder(
+        Guid releaseId,
+        Dictionary<Guid, int> newKeyStatisticOrder)
+    {
+        return await _persistenceHelper.CheckEntityExists<Release>(releaseId)
+            .OnSuccess(_userService.CheckCanUpdateRelease)
+            .OnSuccess<ActionResult, Release, List<KeyStatisticViewModel>>(async release =>
+            {
+                var keyStatistics = _context.KeyStatistics
+                    .Where(ks => ks.ReleaseId == release.Id)
+                    .ToList();
+
+                var idOrderPairList = newKeyStatisticOrder.ToList();
+
+                if (keyStatistics.Count != idOrderPairList.Count)
+                {
+                    return ValidationUtils.ValidationActionResult(
+                        KeyStatsOrderCountShouldEqualReleaseKeyStatsCount);
+                }
```

Comment:
I've switched to using `SequencesAreEqualIgnoringOrder`, and moved that method into `ComparerUtils`.

Even though the `ArgumentException` should never be thrown now, I've left it in - possible it might catch a bug after future refactoring is my thought.

Reply:
Also removed `EnumerableExtensions#IsSameAsIgnoringOrder` - it now uses the common `SequencesAreEqualIgnoringOrder`. Moved and adapted tests for `IsSameAsIgnoringOrder` into `ComparerUtilsTests`.

Reply:
Cool, left a separate comment on `SequencesAreEqualIgnoringOrder`. I think it was a bit more obvious where it was located before that it was only ever being used to compare `Guid` but it now needs restraining to make sure it can only be used with types that are comparable.